### ISSUE TITLE
Investigate different CPU profiling strategies on iOS

### ISSUE THREAD

**Issue by zanderso** [dart-lang/sdk#47139]
Context: https://github.com/flutter/flutter/issues/88466

The strategy of sending SIGPROF conflicts with the need to have lldb attached from the Flutter tool and XCode because lldb intercepts the signal and stops and restarts the process each time, slowing the program down to a crawl.

This issue is to track pursuing other strategies, e.g. using `ptrace()`.

/cc @bkonyi

Side note: there is no `os-ios` GitHub label.

**Comment by a-siva** [comment]
//cc @bkonyi @rmacnak-google

**Comment by a-siva** [comment]
Should this be a P2 @zanderso ?

**Comment by zanderso** [comment]
Users have a workaround of launching the app from the homescreen, and then CPU profiling will work after a `flutter attach`, so this is not blocking. I would say it is lower than P2.

**Comment by a-siva** [comment]
There was the case of some app developers wanting to profile the initial startup latency (e.g: time to first frame etc.), would that be possible with this workaround ?

**Comment by zanderso** [comment]
No, that wouldn't be possible. I think the developers that you're thinking of are primarily debugging their startup issue on Android, however.

**Comment by Jordan-Nelson** [comment]
> Users have a workaround of launching the app from the homescreen, and then CPU profiling will work after a `flutter attach`, so this is not blocking. I would say it is lower than P2.

I am unable to get this to work using the workaround found [here](https://github.com/flutter/flutter/issues/88466#issuecomment-905830680).

On the first attempt it resulted in the exact same scenario as when I had launched the app initially on the iOS device ("No CPU samples record. If you are attempting ..."). On subsequent attempts It hangs trying to attach to the device.

iOS 14.7
flutter 2.2.0

**Comment by ristiisa** [comment]
> Users have a workaround of launching the app from the homescreen, and then CPU profiling will work after a `flutter attach`, so this is not blocking. I would say it is lower than P2.

I was able to use the workaround 2 weeks ago, but not anymore. I tried updating flutter from 2.2.3 -> 2.5.0 but that made no difference... 

What is different is that now dev-tools shows a link to the workaround, that was not there before

**Comment by zanderso** [comment]
Fix re-landed in https://github.com/dart-lang/sdk/commit/db45ae178a71c4df41ceec3c28c6d562dedc132f
